### PR TITLE
docs: add download page

### LINKED ISSUE #10236: Consider adding a "Downloads" page
In https://github.com/apache/datafusion/pull/10233 @tisonkun added a doap file 🙏 

However, one of the fields is a "download" page which they mention doesn't quite map to Rust expectations / common DataFusion usage. 

The situation is somehow complicated due to the existing ASF policies, which make some assumptions in the old days.

Here is the policy: https://infra.apache.org/release-download-pages.html

You can see a formal Download page has a lot of requirements to follow. But for DataFusion, most users just use `cargo add datafusion` and never care if such a page exists.

We recently had [a discussion on the general@incubator.a.o](https://lists.apache.org/thread/m9go306p0ctwjp43ryn6b5hpjbynwcxg) for related topics; while I understand your suggestion here, and we should do what is helpful for our users.

However, Arrow has [a release/download page](https://arrow.apache.org/release/), and I check it meets all the requirements. So if people in DataFusion have time to implement it, we can avoid any further concern.

_Originally posted by @tisonkun in https://github.com/apache/datafusion/pull/10233#discussion_r1579932133_

### PR DESCRIPTION
**Pull request by tisonkun**
## Which issue does this PR close?

Closes #10236

### CODE REVIEW

File: docs/source/download.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Download
+
+The official Apache DataFusion releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),
```
Comment (tisonkun) [inline docs/source/download.md:7]: Once we make the first release, we can update these links and merge this patch.

Comment (tisonkun) [reply docs/source/download.md:26]: image

File: docs/source/download.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  under the License.
+-->
+
+# Download
+
+The official Apache DataFusion releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.45.0](https://www.apache.org/dyn/closer.lua/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz?action=download) ([asc](https://downloads.apache.org/opendal/0.45.0/apache-opendal-0.45.0-src.tar.gz.asc),
```
Comment (andygrove) [inline docs/source/download.md:26]: I assume the opendal references need replacing with datafusion?

Comment (andygrove) [reply docs/source/download.md:26]: Thanks, I missed that. We are planning on creating a new release this week. I am going to start the process tomorrow.

https://github.com/apache/datafusion/issues/10217

Comment (andygrove) [reply docs/source/download.md:26]: We could link to the last release in the arrow site for now?

https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/

**Comment by tisonkun** [comment]
@andygrove Updated. It should be fine to merge now. And perhaps our release manual should contain a step to update the download page with the latest version link.

File: docs/source/download.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+The official Apache DataFusion releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [37.0.0][source-link] ([asc][asc-link],
+[sha512][sha512-link]).
+
+[source-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz?action=download
+[asc-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.asc?action=download
+[sha512-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.sha512?action=download
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply docs/source/download.md:31]: Thank you! I often forget the correct links in use ... 😢

File: docs/source/download.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+The latest source release is [37.0.0][source-link] ([asc][asc-link],
+[sha512][sha512-link]).
+
+[source-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz?action=download
+[asc-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.asc
+[sha512-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.sha512
+
+For older releases, please check the [archive](https://archive.apache.org/dist/datafusion/).
+
+For even older releases when DataFusion is a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```
Comment (phillipleblanc) [inline docs/source/download.md:35]:
```suggestion
For older releases (<= 37) when DataFusion was a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```

Minor rewording

File: docs/source/download.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+## Releases
+
+The latest source release is [37.0.0][source-link] ([asc][asc-link],
+[sha512][sha512-link]).
+
+[source-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz?action=download
+[asc-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.asc
+[sha512-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.sha512
+
+For older releases, please check the [archive](https://archive.apache.org/dist/datafusion/).
```
Comment (phillipleblanc) [inline docs/source/download.md:33]:
```suggestion
For previous releases, please check the [archive](https://archive.apache.org/dist/datafusion/).
```

**Review by phillipleblanc — APPROVED** [review]
Thanks for this! I have some minor rewording suggestions.

**Comment by tisonkun** [comment]
@phillipleblanc Thank you! Integrated.

File: docs/source/download.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+The latest source release is [37.0.0][source-link] ([asc][asc-link],
+[sha512][sha512-link]).
+
+[source-link]: https://www.apache.org/dyn/closer.lua/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz?action=download
+[asc-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.asc
+[sha512-link]: https://downloads.apache.org/arrow/arrow-datafusion-37.0.0/apache-arrow-datafusion-37.0.0.tar.gz.sha512
+
+For previous releases, please check the [archive](https://archive.apache.org/dist/datafusion/).
+
+For older releases (<= 37) when DataFusion was a subproject of Apache Arrow, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```
Comment (comphead) [inline docs/source/download.md:35]:
```suggestion
Note: For releases earlier than 37.0.0 please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```

**Review by comphead — APPROVED** [review]
Thanks @tisonkun 
Minor comments

File: docs/source/download.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Download
+
+The official Apache DataFusion releases are provided as source artifacts.
```
Comment (alamb) [inline docs/source/download.md:22]: I think we may cause confusion if we don't also clarify how the code on crates.io related to the downloads linked from this page

```suggestion
While DataFusion is also distributed via the Rust [crates.io] package manager as a convenience, the 
official Apache DataFusion releases are provided as source artifacts. 

[crates.io]: https://crates.io/crates/datafusion
```

**Review by alamb — APPROVED** [review]
Thanks (again!)  @tisonkun  -- I think it would be good to also mention crates.io (I left a suggestion) but I also think we could fix that as a follow on PR as well

File: docs/source/download.md (hunk 6 above)
Comment (tisonkun) [reply docs/source/download.md:35]:
```suggestion
For releases earlier than 37.0.0, please check [Arrow's archive](https://archive.apache.org/dist/arrow/).
```

**Comment by tisonkun** [comment]
@comphead @alamb Thanks for your suggestions. Integrated.

**Comment by alamb** [comment]
Looks great to me -- thank you @tisonkun @phillipleblanc @andygrove @sebbASF  and @comphead  -- a very nice new page indeed!

**Pull request merged**